dcc-import-gene/src/main/java/org/icgc/dcc/imports/gene/GeneImporter.java

```diff
@@ -57,32 +63,50 @@ public ImportSource getSource() {
     return ImportSource.GENES;
   }
 
+  /**
+   * Main pipeline execution for creating gene model. Calls all readers to pre-compute hashmaps of relevant information
+   * before streaming GTF file for construction of gene model skeleton.
+   */
+  @SneakyThrows
   @Override
   public void execute() {
-    execute(all());
-  }
-
-  @SneakyThrows
-  public void execute(GeneFilter geneFilter) {
     val watch = createStarted();
 
-    log.info("Reading genes BSON stream from {}...", genesBsonUri);
-    val genesBson = readGenesBson();
+    // TODO: Investigate ordering to see if gene.txt does not need to be read twice.
+    log.info("Doing Ensembl Data Joining...");
+    val idMap = IdReader.getIdMap();
+    val nameMap = NameReader.readXrefDisplay();
```

Comment:
Might want to throw a `val watch = Stopwatch.createStarted();` in here to time each of the legs.